nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java

```diff
@@ -274,6 +285,10 @@ private boolean fromUnannotatedPackage(Symbol symbol) {
     public TransferResult<Nullness, NullnessStore<Nullness>> visitBooleanLiteral(
             BooleanLiteralNode booleanLiteralNode,
             TransferInput<Nullness, NullnessStore<Nullness>> input) {
+        ExpressionTree t = booleanLiteralNode.getTree();
+        if (lambdaResultExpression == t) {
+            handler.onDataflowVisitLambdaResultExpression(t, input.getThenStore(), input.getElseStore());
+        }
```

Comment:
I don't love this way of handling single-expression lambdas, but I found no better way. The idea is to call `handler.onDataflowVisitLambdaResultExpression` whenever we visit the `$exp` part of `(x -> $exp)` and `$exp` is a boolean expression (the only one for which it makes sense to consider the then store and else store separately). We don't want to report subexpressions, so we compare by identity against the stored `lambdaResultExpression` reference. Also, not entirely sure how to deal with ternary expressions returning boolean in this case.

Reply:
Yeah let's chat and see if we can come up with a cleaner way to do this.